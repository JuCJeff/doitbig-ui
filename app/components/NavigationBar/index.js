/*
 * navigationBar
 *
 * This is the navigation bar of our App, at the top of each page.
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { Avatar } from '@material-ui/core';
import STS from '../../images/STS.png';
import MenuListComposition from '../Dropdown/index';

const styles = {
  root: {
    flexGrow: 1,
  },
  grow: {
    flexGrow: 1,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
};

function ButtonAppBar(props) {
  const { classes } = props;
  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <Typography variant="h6" color="inherit" className={classes.grow}>
            <Button href="/" color="inherit">
              <Avatar src={STS} />Software Training for Students
            </Button>
          </Typography>
          <Button href="/tracks" color="inherit">
            Tracks
          </Button>
          <Button color="inherit">Courses</Button>
          <MenuListComposition />
        </Toolbar>
      </AppBar>
    </div>
  );
}

ButtonAppBar.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(ButtonAppBar);
