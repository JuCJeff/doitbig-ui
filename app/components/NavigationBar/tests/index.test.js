import React from 'react';
import { shallow } from 'enzyme';
import NavigationBar from '../../../components/NavigationBar/index';

describe('<NavigationBar />', () => {
  it('should render the page message', () => {
    const renderedComponent = shallow(<NavigationBar />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
