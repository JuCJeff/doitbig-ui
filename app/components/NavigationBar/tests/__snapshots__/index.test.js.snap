// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBar /> should render the page message 1`] = `
<ButtonAppBar
  classes={
    Object {
      "grow": "ButtonAppBar-grow-2",
      "menuButton": "ButtonAppBar-menuButton-3",
      "root": "ButtonAppBar-root-1",
    }
  }
/>
`;
